src/inet/TunEndPoint.cpp

```diff
@@ -549,12 +549,12 @@ err_t TunEndPoint::LwIPPostToInetEventQ (struct netif *netif, struct pbuf *p)
     INET_ERROR              err             = INET_NO_ERROR;
     TunEndPoint*            ep              = static_cast<TunEndPoint *>(netif->state);
     Weave::System::Layer&   lSystemLayer    = ep->SystemLayer();
-    PacketBuffer*           buf             = PacketBuffer::NewWithAvailableSize(p->tot_len);
 
     // Starting off with a reserved size of the default WEAVE_SYSTEM_CONFIG_HEADER_RESERVE_SIZE
     // which allows for adding the Weave header and the underlying transport and IP headers
-    // encapsulating this tunneled packet.
+    // encapsulating this tunneled packet. Also need to request space for trailer.
```

Comment:
Recommend modifying the comment as "Allocate space for the tunneled IP packet using the default size for the Weave and underlying TCP/IP headers(WEAVE_SYSTEM_CONFIG_HEADER_RESERVE_SIZE) and the trailer(WEAVE_TRAILER_RESERVE_SIZE) for holding the HMAC".

Reply:
See new commit. I took your suggestions and included some of my own